lms/static/js/groups/views/cohort_discussions_inline.js

```diff
@@ -0,0 +1,142 @@
+        },
+
+        initialize: function (options) {
+            this.template = _.template($('#cohort-discussions-inline-tpl').text());
+            this.cohortSettings = options.cohortSettings;
+        },
+
+        render: function () {
+            var alwaysCohortInlineDiscussions = this.cohortSettings.get('always_cohort_inline_discussions');
+
+            this.$('.cohort-inline-discussions-nav').html(this.template({
+                inlineDiscussionTopics: this.getInlineDiscussions(this.model.get('inline_discussions')),
+                alwaysCohortInlineDiscussions:alwaysCohortInlineDiscussions
+            }));
+
+            $('ul.inline-topics').qubit();
+
+            this.setDisabled(this.$('.cohort-inline-discussions-form .action-save'), true);
+            if (alwaysCohortInlineDiscussions) {
+                this.setDisabled(this.$('.check-discussion-category'), true);
+                this.setDisabled(this.$('.check-discussion-subcategory-inline'), true);
+            }
+        },
+
+        /**
+         Returns the html list for inline discussion topics.
+
+         Args:
+            inlineDiscussions (object): inline discussions object from server
+                with attributes 'entries', 'children' & 'subcategories'.
+
+         Returns:
+            HTML list for inline discussion topics.
+        **/
+        getInlineDiscussions: function (inlineDiscussions) {
+            var categoryTemplate = _.template($('#cohort-discussions-category-tpl').html()),
+                entryTemplate = _.template($('#cohort-discussions-subcategory-tpl').html()),
+                isCategoryCohorted = false,
+                children = inlineDiscussions.children,
+                entries = inlineDiscussions.entries,
+                subcategories = inlineDiscussions.subcategories;
+
+            return _.map(children, function (name) {
+                var html = '', entry;
+                if (entries && _.has(entries, name)) {
+                    entry = entries[name];
+                    html = entryTemplate({
+                        name: name,
+                        id: entry.id,
+                        is_cohorted: entry.is_cohorted,
+                        type: 'inline'
+                    });
+                } else { // subcategory
+                    html = categoryTemplate({
+                        name: name,
+                        entries: this.getInlineDiscussions(subcategories[name]),
+                        isCategoryCohorted: isCategoryCohorted
+                    });
+                }
+                return html;
+            }, this).join('');
+        },
+
+        /**
+         Disables the discussion category checkboxes.
+         Disables the discussion sub-category checkboxes.
+         Enables the save button for inline discussion topics.
+
+         Args:
+            disable (Bool): The flag to enable/disable the elements.
+        **/
+        changeAllInlineDiscussions: function(event) {
```

Comment:
I think `setCohortAllInlineDiscussions` and `setCohortSomeInlineDiscussions` would be more descriptive names.